### PR TITLE
Remove passport-oauth2-jwt-bearer as a dependency

### PR DESCRIPTION
**Pull request by amikheychik**
[passport-oauth2-jwt-bearer](https://github.com/xtuple/passport-oauth2-jwt-bearer) is a micro-package created and published by xTuple. It’s not being supported and the most recent version (0.2.0) contains changes incompatible with version used  (0.1.1).

Actual code is located in one file, so this file is added to the repository directly, so passport-oauth2-jwt-bearer package may be deprecated and later removed.

### CODE REVIEW

File: node-datasource/module/passport-oauth2-jwt-bearer/index.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+/**
+* Framework version.
+*/
+require('pkginfo')(module, 'version');
+
+/**
+* Expose constructors.
+*/
+exports.Strategy = Strategy;
```
Comment (bendiy) [inline node-datasource/module/passport-oauth2-jwt-bearer/index.js]: Add new line at end of file.

**Comment by bendiy** [comment]
CI is failing at:

```
> mocha -R spec test/database/*

<<ERROR 2018-10-24T10:23:59.806Z>> Unrecoverable exception. Unexpected token }/home/xtuple/jenkins/workspace/xtuple-pr/xtuple/test/database/functions/applyarcreditmemotobalance.js:76
});
^

SyntaxError: Unexpected token }
    at createScript (vm.js:80:10)
    at Object.runInThisContext (vm.js:139:10)
    at Module._compile (module.js:616:28)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Module.require (module.js:596:17)
    at require (internal/module.js:11:18)
    at /home/xtuple/jenkins/workspace/xtuple-pr/xtuple/node_modules/mocha/lib/mocha.js:231:27
    at Array.forEach (<anonymous>)
    at Mocha.loadFiles (/home/xtuple/jenkins/workspace/xtuple-pr/xtuple/node_modules/mocha/lib/mocha.js:228:14)
    at Mocha.run (/home/xtuple/jenkins/workspace/xtuple-pr/xtuple/node_modules/mocha/lib/mocha.js:536:10)
    at Object.<anonymous> (/home/xtuple/jenkins/workspace/xtuple-pr/xtuple/node_modules/mocha/bin/_mocha:582:18)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Function.Module.runMain (module.js:693:10)
    at startup (bootstrap_node.js:191:16)
    at bootstrap_node.js:612:3
+ npm run-script jshint

> xtuple@4.12.0-alpha jshint /home/xtuple/jenkins/workspace/xtuple-pr/xtuple
> jshint lib/backbone-x/source lib/enyo-x/source lib/tools/source enyo-client/application/source/ enyo-client/extensions node-datasource

node-datasource/module/passport-oauth2-jwt-bearer/strategy.js: line 38, col 30, ['assertion'] is better written in dot notation.
node-datasource/module/passport-oauth2-jwt-bearer/strategy.js: line 43, col 27, ['assertion'] is better written in dot notation.

2 errors
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! xtuple@4.12.0-alpha jshint: `jshint lib/backbone-x/source lib/enyo-x/source lib/tools/source enyo-client/application/source/ enyo-client/extensions node-datasource`
npm ERR! Exit status 2
npm ERR! 
npm ERR! Failed at the xtuple@4.12.0-alpha jshint script.
```

**Comment by bendiy** [comment]
https://github.com/xtuple/xtuple/commit/517dc184f4d077aa089630fa97bd912ec007fd99#r31124368

**Comment by bendiy** [comment]
jenkins - retest this please

**Comment by bendiy** [comment]
jenkins - retest this please

**Comment by amikheychik** [comment]
I'm closing this request, as this dependency will not be needed at all: in the new REST server that is being done in `xtuple/xdruple-extension` `passport` module is not used at all.

**Pull request closed without merging**